server/src/api/v1.py

```diff
@@ -101,7 +100,33 @@ def has_attachments(data: dict) -> bool:
     )
 
 
-def job_builder(data):
+def check_token_permission(
+    auth_token: str, secret_key: str, priority: int, queue: str
+) -> bool:
+    """
+    Validates token received from client and checks if it can
+    push a job to the queue with the requested priority
+    """
+    if auth_token is None:
+        abort(401, "No authentication token specified")
+    else:
+        try:
+            decoded_jwt = jwt.decode(
+                auth_token,
+                secret_key,
+                algorithms="HS256",
+                options={"require": ["exp", "iat", "sub", "max_priority"]},
```

Comment:
what about client permissions that don't have a max_priority? We already know of at least one other use we want to add - for restricted queues. It's entirely possible that they have `allowed_queues` or some other future key, but not `max_priority`.